src/agent/agent.c

```diff
@@ -1805,47 +1805,51 @@ static bool ensure_orch_address(Agent *agent) {
         if (agent->orch_addr != NULL) {
                 return true;
         }
-
-
         if (agent->host == NULL) {
                 hirte_log_errorf("No manager host specified for agent '%s'", agent->name);
                 return false;
         }
 
+        char *ip_address = agent->host;
+        bool host_is_ipv4 = isIPv4Addr(ip_address);
+        bool host_is_ipv6 = isIPv4Addr(ip_address);
 
-        if (!isIPv4Addr(agent->host) && !isIPv6Addr(agent->host)) {
-                char *ip_address = malloc0_array(0, sizeof(char), INET6_ADDRSTRLEN);
-                int r = get_address(agent->host, &ip_address);
-                if (r != 0) {
-                        hirte_log_errorf("Failed to get IP address from: '%s'", agent->host);
+        _cleanup_free_ char *resolved_ip_address = NULL;
+        if (!host_is_ipv4 && !host_is_ipv6) {
+                int r = get_address(ip_address, &resolved_ip_address);
+                if (r < 0) {
+                        hirte_log_errorf(
+                                        "Failed to get IP address from host '%s': %s",
+                                        agent->host,
+                                        strerror(-r));
                         return false;
                 }
-                hirte_log_infof("Translated '%s' to '%s'", agent->host, ip_address);
-                copy_str(&agent->host, ip_address);
+                hirte_log_infof("Translated '%s' to '%s'", ip_address, resolved_ip_address);
+                ip_address = resolved_ip_address;
         }
 
-        // IPv4
-        if (isIPv4Addr(agent->host)) {
+        if (host_is_ipv4 || isIPv4Addr(ip_address)) {
```

Comment:
I don't think we need host_is_ipv4 here as we need a new validation anyway with isIPv4Addr(ip_address).

Reply:
If `host_is_ipv4` is true for the original `agent->host`, then we didn't resolve a domain to ip and don't need to do a `isIPv4Addr` again. The or ensures that we only execute `isIPv4Addr` when we resolved a domain.